src/Hosting/Hosting/src/Internal/HostingMetrics.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Metrics;
+
+namespace Microsoft.AspNetCore.Hosting;
+
+internal sealed class HostingMetrics : IDisposable
+{
+    private readonly Meter _meter;
+    private readonly Counter<long> _totalRequestsCounter;
+    private readonly UpDownCounter<long> _currentRequestsCounter;
+    private readonly Counter<long> _failedRequestsCounter;
+    private readonly Histogram<double> _requestTime;
+
+    public HostingMetrics(IMeterFactory metricsFactory)
+    {
+        _meter = metricsFactory.CreateMeter("Microsoft.AspNetCore.Hosting");
+
+        _totalRequestsCounter = _meter.CreateCounter<long>(
+            "total-requests",
+            description: "Total Requests");
+
+        _currentRequestsCounter = _meter.CreateUpDownCounter<long>(
+            "current-requests",
+            description: "Current Requests");
+
+        _failedRequestsCounter = _meter.CreateCounter<long>(
+            "failed-requests",
+            description: "Failed Requests");
+
+        _requestTime = _meter.CreateHistogram<double>(
```

Comment:
How expensive is this and how useful by default without custom buckets?

Reply:
What do you mean by custom buckets? This is the `CreateHistogram` create method: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter.createhistogram?view=net-7.0

Reply:
I'm guessing @davidfowl is refering to the fact that the API doesn't allow the histogram creator to specify the cutoff points that separate the histogram buckets so you are at the mercy of whatever the aggregator has decided to do. OTel uses a pretty sparse set of buckets by default but it is [configurable](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/customizing-the-sdk/README.md#specify-custom-boundaries-for-histogram). Anything using MetricsEventSource (dotnet-counters, dotnet-monitor) uses a much denser exponential histogram where no configuration has yet been exposed via a public interface. At present MetricsEventSource guarantees calculated percentiles are [accurate](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Metrics/ExponentialHistogramAggregator.cs,17) to within +- 0.1% of the true value you would get by storing and sorting every data point.